src/Datastore/DatastoreSessionHandler.php

```
@@ -0,0 +1,197 @@
+ * Custom session handler backed by Cloud Datastore.
+ *
+ * {@see http://php.net/manual/en/class.sessionhandlerinterface.php}
+ *
+ * It uses the session.save_path as the Datastore namespace for isolating the
+ * session data from your application data. It replaces '/' with '_' for the
+ * namespaceId.
+ * It also uses session.name as the Datastore kind. It uses the session id as
+ * the Datastore id. By default, it does nothing on gc for reducing the cost.
+ * Pass positive value up to 1000 for $gcLimit parameter to delete entities
+ * in gc.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Datastore\DatastoreClient;
+ * use Google\Cloud\Datastore\DatastoreSessionHandler;
+ *
+ * $handler = new DatastoreSessionHandler(new DatastoreClient());
+ * session_set_save_handler($handler, true);
+ * session_start();
+ *
+ * # Then read and write the $_SESSION array.
+ * ```
+ */
+class DatastoreSessionHandler implements \SessionHandlerInterface
+{
+    const DEFAULT_GC_LIMIT = 0;
+    const DEFAULT_KIND = 'PHPSESSID';
+    const DEFAULT_NAMESPACE_ID = 'sessions';
+
+    /* @var int */
+    private $gcLimit;
+
+    /* @var DatastoreClient */
+    private $datastore;
+
+    /* @var string */
+    private $kind;
+
+    /* @var string */
+    private $namespaceId;
+
+    /* @var Transaction */
+    private $transaction;
+
+    /**
+     * @param DatastoreClient $datastore Datastore client
+     * @param int $gcLimit A number of entities to delete in garbage collection
+     */
+    public function __construct(
+        DatastoreClient $datastore,
+        $gcLimit = self::DEFAULT_GC_LIMIT
+    ) {
+        $this->datastore = $datastore;
+        // Cut down to 1000
+        $this->gcLimit = min($gcLimit, 1000);
+        $this->kind = self::DEFAULT_KIND;
+        $this->namespaceId = self::DEFAULT_NAMESPACE_ID;
+    }
+
+    public function open($savePath, $sessionName)
+    {
+        $this->kind = $sessionName;
+        $this->namespaceId = str_replace('/', '_', $savePath);
```

This seems like a potential source of surprise and confusion to me. Would it be better to throw an exception rather than silently transforming the value the user provides?